packages/devextreme/testing/tests/DevExpress.ui.widgets/multiView.tests.js

```diff
@@ -504,6 +504,36 @@ QUnit.module('interaction via swipe', {
         });
     });
 
+    QUnit.test('there should not be infinite loop if try swiping when all items are hidden', function(assert) {
+        const $multiView = $('#multiView').dxMultiView({
+            items: [
+                { text: '1', visible: false },
+                { text: '2', visible: false },
+                { text: '3', visible: false },
+            ],
+            selectedIndex: 2,
+            loop: true,
+            swipeEnabled: true
+        });
+        const instance = $multiView.dxMultiView('instance');
+        const pointer = pointerMock($multiView);
+
+        const isItemVisibleStub = sinon.stub(instance, '_isItemVisible').callsFake(() => {
+            if(isItemVisibleStub.callCount === 20) {
+                assert.ok(false, 'infinite loop detected');
+                return true;
+            }
+            return false;
+        });
+
+        pointer.start().swipeStart().swipe(-0.5).swipeEnd(-1);
+
+
+        assert.ok(true, 'no infinite loop');
+
+        isItemVisibleStub.restore();
```

Comment:
I would prefer to be safe and, if something went wrong, revert to the state before the test execution
```
const isItemVisibleStub = sinon.stub(...);

try {
    // code
} finally {
    isItemVisibleStub.restore();
}
```

Reply:
[4938424](https://github.com/DevExpress/DevExtreme/pull/28299/commits/493842406d6cb591fd111d55fee5de9c62515e68)